Backport of Rockchip/NanoPi enhancements (closed in favor of #18347)

dmsza opened the pull request:
**Update**: closed in favor of #18347 (due to an origin branch renaming and housekeeping)

~Backport of 4 commits from PR [17638](https://github.com/openwrt/openwrt/pull/17638) by @lespinasse (cherry-picked from main).~

~I am running this PR for about 3 weeks merged into 24.10.0 in a NanoPi R6S and no issues found.~

lespinasse committed:
rockchip: show boot stages on nanopi R6 system LED

Set up openwrt to show boot progress on the nanopi R6S or R6C system LED.

The LED blinking states indicate the boot stage. The LED is defined as
a power LED, but can still be set to heartbeat in /etc/config/system
after the system is done booting.

Signed-off-by: Michel Lespinasse <michel@lespinasse.org>
Link: https://github.com/openwrt/openwrt/pull/17638
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit de9ad11dac094b6074c9b23341acbc28806404ae)

lespinasse committed:
rockchip: configure reset button on nanopi R6

Set up openwrt to handle the reset button appropriately (so that it
can trigger the various recovery modes) on the nanopi R6S and R6C models.

Signed-off-by: Michel Lespinasse <michel@lespinasse.org>
Link: https://github.com/openwrt/openwrt/pull/17638
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 2e7d060fd1ba6160067df1a2ba303c7c2efe2a7e)

lespinasse committed:
rockchip: set network IRQ affinity to fast CPU cores

The nanopi R6S, R6C and nanopc T6 platforms are based on rk3588(s) SoC,
which has fast and slow CPU cores. Set up network interrupt affinity to be
on the fast CPU cores by default. This is similar to the way this was
already configured on nanopi R4S.

Signed-off-by: Michel Lespinasse <michel@lespinasse.org>
Link: https://github.com/openwrt/openwrt/pull/17638
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 59215154a0eba77f0a79a6e30069a3037603ebf2)

lespinasse committed:
kernel: load r8169 network module at boot time

This allows the network interface naming to be stable, free from any
possible interaction from external USB network devices that might
claim usb* interface names.

(This was a real problem I encountered with a nanopi R6S device and
an external rtl8152 usb3 network controller - the USB controller would
claim the eth1 name, causing much confusion).

Signed-off-by: Michel Lespinasse <michel@lespinasse.org>
Link: https://github.com/openwrt/openwrt/pull/17638
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

lespinasse commented:
Thank you. I saw my PR 17638 being approved late last night, and thought I should submit it against 24.10 branch too, so I'm happy you beat me to it. I think the changes are small enough for 24.10, so I hope openwrt team will approve this pull request.

dmsza committed:
Merge branch 'openwrt:openwrt-24.10' into openwrt-24.10

lespinasse commented:
The recent merge kinda obscures that, but clicking the checkmark next to the fd75f69 commit above still shows that all tests passed (208 successful and 2 skipped).

TLarg committed:
rockchip: fix missing ".patch" file extension

Fixes missing .patch file extension in:
- https://github.com/openwrt/openwrt/commit/de9ad11dac094b6074c9b23341acbc28806404ae
- https://github.com/openwrt/openwrt/commit/2e7d060fd1ba6160067df1a2ba303c7c2efe2a7e

Signed-off-by: Tomas Lara <tl849670@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18281
Signed-off-by: Robert Marko <robimarko@gmail.com>
(cherry picked from commit f86476b18bb99f9f9491c30430bfa9fa110cb394)

dmsza committed:
Merge branch 'openwrt:openwrt-24.10' into openwrt-24.10

dmsza commented:
@lespinasse please let me know if you prefer to create a new PR on your account, and if so I can close this one.

lespinasse commented:
I don't have a preference, do you want me to ? My guess is switching the pull request mid-flight would only confuse openwrt team.

dmsza commented:
OK, let's keep this PR then!

dmsza commented:
@lespinasse while doing a house keeping in my repo, I renamed the origin branch of this PR to "openwrt-24.10-rockchip-enhancements)" with the assumption that GitHub would keep the PR referenced to the renamed one.

But no, it seems that GitHub uses the literal branch name in PRs and does not automatically update PRs (so it handles a branch rename as branch deleted in the referenced PRs, lesson learned).

This way this PR was automatically closed, and a new one will be required. Since you was the author of these enhancements, I believe that perhaps it would best that you could recreate the backport PR.

dmsza commented:
@lespinasse while doing a house keeping in my repo, I renamed the origin branch of this PR to "openwrt-24.10-rockchip-enhancements)" with the assumption that GitHub would keep the PR referenced to the renamed one.

But no, it seems that GitHub uses the literal branch name in PRs and does not automatically update PRs (so it handles a branch rename as branch deleted in the referenced PRs, lesson learned).

This way this PR was automatically closed. I am a little newbie regarding GitHub, and I am trying to change this PR to point to the renamed branch.

dmsza commented:
@lespinasse I went ahead I created a new PR (#18347) that supersedes this new one after my repo housekeeping messed with this one. I will keep now the source branch untouched until the new PR is merged into 24.10.
